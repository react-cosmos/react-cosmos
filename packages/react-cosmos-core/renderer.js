module.exports = require('./dist/renderer/index.js');
