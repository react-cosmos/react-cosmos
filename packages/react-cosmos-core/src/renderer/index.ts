export * from './FixtureLoader/getDecoratedFixtureElement/index.js';
export * from './getFixtureList.js';
export * from './getSortedDecoratorsForFixturePath.js';
export * from './reactTypes.js';
export * from './registerPlaygroundShortcuts.js';
export * from './rendererConfig.js';
export * from './types.js';
