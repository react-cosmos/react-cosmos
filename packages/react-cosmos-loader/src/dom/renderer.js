// @flow

import { render, unmountComponentAtNode } from 'react-dom';

import type { LoaderWebOpts } from 'react-cosmos-flow/loader';
import type { Renderer } from 'react-cosmos-flow/context';

export function createDomRenderer(opts?: LoaderWebOpts = {}): Renderer {
  const { containerQuerySelector } = opts;
  const container = getDomContainer(containerQuerySelector);

  return function renderer(element) {
    render(element, container);

    return {
      unmount() {
        unmountComponentAtNode(container);
      },
      toJSON: () => null
    };
  };
}

function getDomContainer(querySelector?: string): HTMLElement {
  if (!querySelector) {
    return createDomContainer();
  }

  const existingContainer = document.querySelector(querySelector);
  if (!existingContainer) {
    console.warn(
      `[Cosmos] Could not find ${querySelector} element. Created fresh DOM container.`
    );
    return createDomContainer();
  }

  return existingContainer;
}

function createDomContainer() {
  const existingNode = document.getElementById('root');
  if (existingNode) {
    return existingNode;
  }

  const newNode = document.createElement('div');
  newNode.setAttribute('id', 'root');
  if (document.body) {
    document.body.appendChild(newNode);
  }

  return newNode;
}
