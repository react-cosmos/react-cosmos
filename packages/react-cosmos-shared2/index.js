// @flow

exports.RENDERER_ID = require('./dist/index').RENDERER_ID;

exports.updateItem = require('./dist/utility').updateItem;
exports.replaceOrAddItem = require('./dist/utility').replaceOrAddItem;
exports.removeItem = require('./dist/utility').removeItem;

exports.extractValuesFromObject = require('./dist/values').extractValuesFromObject;
exports.areValuesEqual = require('./dist/values').areValuesEqual;

exports.updateState = require('./dist/state').updateState;

exports.getFixtureStateProps = require('./dist/fixtureState').getFixtureStateProps;
exports.getFixtureStatePropsInst = require('./dist/fixtureState').getFixtureStatePropsInst;
exports.updateFixtureStateProps = require('./dist/fixtureState').updateFixtureStateProps;
exports.getFixtureStateState = require('./dist/fixtureState').getFixtureStateState;
exports.getFixtureStateStateInst = require('./dist/fixtureState').getFixtureStateStateInst;
exports.updateFixtureStateState = require('./dist/fixtureState').updateFixtureStateState;
