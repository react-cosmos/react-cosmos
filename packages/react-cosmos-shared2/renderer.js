module.exports = require('./dist/renderer');
