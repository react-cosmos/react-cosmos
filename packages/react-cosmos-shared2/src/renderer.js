// @flow

export const RENDERER_MESSAGE_EVENT_NAME = 'cosmos-cmd';
