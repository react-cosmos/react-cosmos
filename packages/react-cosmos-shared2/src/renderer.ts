import { FixtureState } from './fixtureState';

// FYI: Renderer ids are self assigned in remote environments, so uniqueness
// cannot be established by consensus
export type RendererId = string;

export type FixtureNamesByPath = {
  [fixturePath: string]: null | string[];
};

export type FixtureId = {
  path: string;
  name: null | string;
};

export type PingRenderersRequest = {
  type: 'pingRenderers';
};

export type SelectFixtureRequest = {
  type: 'selectFixture';
  payload: {
    rendererId: RendererId;
    fixtureId: FixtureId;
    fixtureState: null | FixtureState;
  };
};

export type UnselectFixtureRequest = {
  type: 'unselectFixture';
  payload: {
    rendererId: RendererId;
  };
};

export type SetFixtureStateRequest = {
  type: 'setFixtureState';
  payload: {
    rendererId: RendererId;
    // The fixture ID is sent alongside the fixture state change to ensure
    // that the fixture state is only paired with its corresponding fixture
    fixtureId: FixtureId;
    fixtureState: null | FixtureState;
  };
};

export type RendererRequest =
  | PingRenderersRequest
  | SelectFixtureRequest
  | UnselectFixtureRequest
  | SetFixtureStateRequest;

export type OnRendererRequest = (msg: RendererRequest) => unknown;

export type RendererReadyResponse = {
  type: 'rendererReady';
  payload: {
    rendererId: RendererId;
    fixtures: FixtureNamesByPath;
  };
};

export type RendererErrorResponse = {
  type: 'rendererError';
  payload: {
    rendererId: RendererId;
  };
};

export type FixtureListUpdateResponse = {
  type: 'fixtureListUpdate';
  payload: {
    rendererId: RendererId;
    fixtures: FixtureNamesByPath;
  };
};

// Caused by an organic state change inside the renderer. Also dispatched
// after a fixtureSelect request, when rendering stateful components, as their
// initial state is read.
export type FixtureStateChangeResponse = {
  type: 'fixtureStateChange';
  payload: {
    rendererId: RendererId;
    // The fixture ID is sent alongside the fixture state to ensure that the
    // fixture state is only paired with its corresponding fixture
    fixtureId: FixtureId;
    // Entire fixture state is included
    fixtureState: null | FixtureState;
  };
};

export type RendererResponse =
  | RendererReadyResponse
  | RendererErrorResponse
  | FixtureListUpdateResponse
  | FixtureStateChangeResponse;

export type OnRendererResponse = (msg: RendererResponse) => unknown;

export const RENDERER_MESSAGE_EVENT_NAME = 'cosmos-cmd';
