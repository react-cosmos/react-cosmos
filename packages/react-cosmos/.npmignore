src
babel.config.js
tsconfig.build.json
