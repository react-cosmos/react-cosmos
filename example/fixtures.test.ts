import { FixtureId } from 'react-cosmos-shared2/renderer';
import { getCosmosConfigAtPath, getFixtures } from 'react-cosmos';
import { create } from 'react-test-renderer';

const cosmosConfig = getCosmosConfigAtPath(require.resolve('./cosmos.config'));

it('returns fixture URLs', async () => {
  expect.hasAssertions();
  const fixtures = await getFixtures({ cosmosConfig });
  fixtures.forEach(({ fixtureId, getElement }) => {
    const renderer = create(getElement());
    expect(renderer.toJSON()).toMatchSnapshot(stringifyFixtureId(fixtureId));
  });
});

function stringifyFixtureId(fixtureId: FixtureId) {
  const { path, name } = fixtureId;
  return name ? `${path} ${name}` : path;
}
