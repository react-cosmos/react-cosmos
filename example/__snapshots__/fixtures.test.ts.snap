// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns fixture elements: __fixtures__/Hello World.ts 1`] = `"Hello World!"`;

exports[`returns fixture elements: __fixtures__/Props Playground.tsx 1`] = `
<pre>
  {
  "string": "How are you doing?",
  "number": 1989,
  "boolean": false,
  "object": {
    "string": "Pretty good can't complain",
    "number": 1337,
    "object": {
      "number": 555
    },
    "null": null,
    "unserializable": {}
  }
}
</pre>
`;

exports[`returns fixture elements: __fixtures__/Values Playground.tsx 1`] = `
<pre>
  {
  "string": "How are you doing?",
  "number": 1989,
  "boolean": true,
  "object": {
    "isAdmin": true,
    "name": "Pat D",
    "age": 44
  },
  "array": [
    {
      "isAdmin": true,
      "name": "Pat D",
      "age": 44
    },
    {
      "isAdmin": false,
      "name": "Dan B",
      "age": 39
    }
  ]
}
</pre>
`;

exports[`returns fixture elements: Counter/index.fixture.tsx default 1`] = `
<button
  onClick={[Function]}
>
  0
   
  times
</button>
`;

exports[`returns fixture elements: Counter/index.fixture.tsx large number 1`] = `
<button
  onClick={[Function]}
>
  555555555
   
  times
</button>
`;

exports[`returns fixture elements: Counter/index.fixture.tsx small number 1`] = `
<button
  onClick={[Function]}
>
  5
   
  times
</button>
`;

exports[`returns fixture elements: CounterButton/index.fixture.tsx 1`] = `
<button
  onClick={[Function]}
>
  0
   
  times
</button>
`;

exports[`returns fixture elements: WelcomeMessage/index.fixture.tsx 1`] = `
<div
  style={
    Object {
      "border": "2px dashed #AAA",
      "borderRadius": 8,
      "padding": "0 16px",
    }
  }
>
  <h1>
    Hi
     
    <strong>
      Maggie
    </strong>
    !
  </h1>
</div>
`;
